guide/src/class/protocols.md

```diff
@@ -1,12 +1,14 @@
 ## Class customizations
 
-Python's object model defines several protocols for different object behavior, like sequence,
-mapping or number protocols. PyO3 defines separate traits for each of them. To provide specific
-Python object behavior, you need to implement the specific trait for your struct. Important note,
-each protocol implementation block has to be annotated with the `#[pyproto]` attribute.
+Python's object model defines several protocols for different object behavior, like sequence, mapping or number protocols. You may be familiar with implementing these protocol in Python classes by "dunder" methods, such as `__str__` or `__repr__`.
 
-All `#[pyproto]` methods which can be defined below can return `T` instead of `PyResult<T>` if the
-method implementation is infallible. In addition, if the return type is `()`, it can be omitted altogether.
+In the Python C-API which PyO3 is dependent upon, many of these protocol methods have to be provided into special "slots" on the class type object. To fill these slots PyO3 uses the `#[pyproto]` attribute in combination with special traits.
+
+The special traits are listed in this chapter of the guide. See also the [documentation for the `pyo3::class` module]({{#PYO3_DOCS_URL}}/pyo3/class/index.html).
```

Comment:
Here, we explain Python's type slots first and then PyO3's pyproto. I propose to reverse the order and note that we have traits and pyproto for some special dunder methods first. We can add the reasons later.